addon/model/util/gen-tree-walker.ts

```diff
@@ -394,6 +389,7 @@ export default class GenTreeWalker<T extends Walkable = Walkable> {
       }
       result = this.filterNode(node);
       if (result === FilterResult.FILTER_ACCEPT) {
+        if (node === fromNode) return null;
```

Comment:
I would assume that this check can be moved 2 lines up, can you try that? 
I think the check is also correct here, but it hinges on the fact that if the result is NOT accept, we will also return null. 
Not that this is something I can see ever changing, but it may be more robust to not rely on that?
If you move it and it works, consider the pr approved and just merge.

Reply:
ill just leave it as it is in the interest of time